tests/foreman/ui/test_reporttemplates.py

```diff
@@ -345,3 +403,60 @@ def test_negative_nonauthor_of_report_cant_download_it(session):
     :expectedresults: Report can't be downloaded. Error.
     :CaseImportance: High
     """
+
+
+@tier3
+def test_positive_generate_entitlements_reports_multiple_formats(session, content_setup):
+    """Generate reports using the Entitlements template in html, yaml, json, and csv format.
+
+        :id: b268663d-c213-4e59-8f81-61bec0838b1e
+
+
+        :setup: Installed Satellite with Organization, Activation key,
+                Content View, Content Host, and Subscriptions.
+
+        :steps:
+            1. Monitor -> Report Templates
+            2. Entitlements -> Generate
+            3. Click the dropdown to select output format
+            4. Submit
+
+        :expectedresults: reports are generated containing all the expected information
+                          regarding Entitlements for each output format.
+
+        :CaseImportance: High
+    """
+    with VirtualMachine(distro=DISTRO_RHEL7) as vm:
+        vm.install_katello_ca()
+        vm.register_contenthost(content_setup[0].label, content_setup[1].name)
+        assert vm.subscribed
+        with session:
+            session.organization.select(content_setup[0].name)
+            session.location.select('Default Location')
+            result_json = session.reporttemplate.generate("Entitlements",
+                                                          values={'output_format': 'JSON'})
+            with open(result_json) as json_file:
+                data_json = json.load(json_file)
+                assert data_json[0]['Name'] == vm.hostname
+                assert data_json[0]['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
+            result_yaml = session.reporttemplate.generate("Entitlements",
+                                                          values={'output_format': 'YAML'})
+            with open(result_yaml) as yaml_file:
+                data_yaml = yaml.load(yaml_file, Loader=yaml.FullLoader)
+                assert data_yaml[0]['Name'] == vm.hostname
+                assert data_yaml[0]['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
+            result_csv = session.reporttemplate.generate("Entitlements",
+                                                         values={'output_format': 'CSV'})
+            with open(result_csv) as csv_file:
+                data_csv = csv.DictReader(csv_file)
+                res = next(data_csv)
+                assert res['Name'] == vm.hostname
+                assert res['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
+            result_html = session.reporttemplate.generate("Entitlements",
+                                                          values={'output_format': 'HTML'})
+            with open(result_html) as html_file:
+                parser = etree.HTMLParser()
+                tree = etree.parse((html_file), parser)
```

Comment:
I guess it makes sense to leave it as is for consistency, but `etree.parse()` can take path to file as first argument.

Either way, there is no need for that argument to be in parentheses.